src/backend/distributed/utils/metadata_cache.c

```diff
@@ -2558,6 +2605,8 @@ InitializeWorkerNodeCache(void)
 		workerNode->nodeRole = currentNode->nodeRole;
 		strlcpy(workerNode->nodeCluster, currentNode->nodeCluster, NAMEDATALEN);
 
+		WorkerNodeArray[workerNodeIndex++] = workerNode;
```

Comment:
It makes me nervous to iteratively build a global variable instead of atomically swapping it from one consistent state to another but I suppose we do the same thing with `WorkerNodeHash` so this isn't any worse than the function already way.